README.md

```diff
@@ -170,6 +171,24 @@ There’s [`babel-plugin-styled-components`](https://github.com/styled-component
 
 **Migrate to `unfetch` with caution.** While it supports the most popular API parts, your app might break if it relies on something less common.
 
+## Unspecific Libraries
+
+Of course, there are also optimization tips for other libaries too. You can use them with common sense to get smaller / more performant bundles.
+
+### `babel-plugin-transform-imports`
+
+> ✅ Safe to use by default / [How to enable](https://www.npmjs.com/package/babel-plugin-transform-imports) / Added by [@kurtextrem](https://twitter.com/kurtextrem)
+
+This handy babel plugin will transform your imports to only import specific components, which ensures not the whole library gets included (if tree-shaking is missing or unavailable).
```

Comment:
> if tree-shaking is missing or unavailable

I’d suggest to rephrase it – probably to something like “if tree-shaking is ineffective for this library”, or to what better tastes you. Below, I’m explaining why.

Generally, tree-shaking can’t be _missing or unavailable_ (at least, in ES modules). What _instead_ happens and why _instead_ this plugin is helpful is the following:

1\. `Grid`, `Row` and `Col` are originally imported from `react-bootstrap/src/Grid.js`. Every component in these files is a class:

```js
// react-bootstrap/src/Grid.js
class Grid extends React.Component { }
```

2\. When the package is published, each class is compiled with Babel. Babel converts these classes into IIFEs (immediately-invoked function expressions) ([see yourself the repl](https://babeljs.io/repl/#?babili=false&browsers=&build=&builtIns=false&code_lz=MYGwhgzhAEDiBOBLAJtA3gKGsg9gFQAtEIAKASnS2miwF8Nag&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=6.26.0&envVersion=)):

```js
// react-bootstrap/lib/Grid.js (approximate code)
var Grid = (function () {
  // ...
})();
```

3\. You import `Grid`, `Row` and `Col`. Webpack+UglifyJS try to tree-shake non-imported modules, but they can’t remove them! In step 2, all these classes were converted into IIFEs, and webpack + UglifyJS don’t know if these IIFEs have any side effects. Therefore, webpack + UglifyJS can’t be sure if removing these IIFEs is safe.

(Note: I’m writing “webpack + UglifyJS” because, to decrease the bundle with tree-shaking, both of them [need to be present](https://developers.google.com/web/fundamentals/performance/webpack/decrease-frontend-size#use_es_modules).)

4\. This is where the plugin comes into play. It replaces the full import with specific imports, and other classes aren’t even touched. Thanks to this, the bundle gets smaller.

Side note: in webpack 4, this can be solved on the library’s side by setting `sideEffects: false` in library’s `package.json`. Lodash (or lodash-es, I don’t remember) already does this.

Reply:
Also babel@7 annotates those class IIFEs with `#__PURE__` annotations that should help uglifyJS to dead code eliminate them in the future. Unfortunately there is also still unresolved issue of static class properties being assigned outside of the IIFE and thus preventing tree-shaking in those cases where you use static properties.

As temporary solution you can also use https://github.com/Andarist/babel-plugin-annotate-pure-calls plugin to annotate all initialization calls that are placed in assignment contexts, but it must be applied carefully if you want to use it in applications.